streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableOuterJoin.java

```diff
@@ -126,16 +147,16 @@ public void init(final ProcessorContext context) {
         }
 
         @Override
-        public R get(final K key) {
+        public ValueAndTimestamp<R> get(final K key) {
             R newValue = null;
-            final V1 value1 = valueGetter1.get(key);
-            final V2 value2 = valueGetter2.get(key);
+            final V1 value1 = getValueOrNull(valueGetter1.get(key));
+            final V2 value2 = getValueOrNull(valueGetter2.get(key));
 
             if (value1 != null || value2 != null) {
                 newValue = joiner.apply(value1, value2);
             }
 
-            return newValue;
+            return ValueAndTimestamp.make(newValue, -1L);
```

Comment:
Interesting. Why is this different than the other joins, where we look up both getter timestamps and compute the max?

Reply:
Ups. This seems wrong... Guess I put a surrogate `-1` to make it compile at some point but forget to fix it...